### ISSUE TITLE
[BUG] Inactive UiStepper steps violate a11y contrast rules

### ISSUE THREAD

**Issue by kubajmarek** [infermedica/component-library#222]
[UiStepper](https://main--639728541e2039caadfe5404.chromatic.com/iframe.html?args=&id=molecules-stepper--with-current-step&viewMode=story) now uses UiText instead of UiButtons for non-navigable steps, with `--color-text-disabled`. Unfortunately, on our background (`--color-background-subtle`), the contrast between the two is too low:

<img width="848" alt="Screenshot 2023-06-14 at 10 57 56" src="https://github.com/infermedica/component-library/assets/24963770/4986d8f2-169b-48a3-b9d1-35fc0e653369">

Previously this wasn't an issue as those steps were disabled buttons, which don't have to obey the contrast rules.
